### PR TITLE
Update orchestra.py

### PR DESCRIPTION
**Pull request by chamaka1**
website and overlay will get message of plan generated when a plan is generated and when power model starts. They need to subscribe to power_model/plan_name

### CODE REVIEW

File: orchestra.py [hunk 1]
```diff
@@ -83,10 +81,11 @@ def on_message(client, userdata, msg):
...
             interps_loaded = pickle.load(y)
             
-        zone_confirmation = str(interps_loaded[2]).strip('[]') 
+        zone_confirmation = str(interps_loaded[2]) 
        
-        client.publish("power_model/zone_confirmation", zone_confirmation)
-        
+        plan_name_message = "plan_name= " + str(plan_name) + "-" + zone_confirmation
+        client.publish("power_model/plan_name", plan_name_message)
+        client.publish("power_model/plan_generated", plan_name_message)
```
Comment (khanguslee) [inline orchestra.py]: Seems like the topic `power_model/plan_generated` has published twice in this section. The body of the message also is different. What does this topic do?

File: orchestra.py [hunk 2]
```diff
@@ -83,10 +81,11 @@ def on_message(client, userdata, msg):
         with open(zone_interpolation_file_path, 'rb') as y:
             interps_loaded = pickle.load(y)
             
-        zone_confirmation = str(interps_loaded[2]).strip('[]') 
+        zone_confirmation = str(interps_loaded[2]) 
        
-        client.publish("power_model/zone_confirmation", zone_confirmation)
-        
+        plan_name_message = "plan_name= " + str(plan_name) + "-" + zone_confirmation
```
Comment (khanguslee) [inline orchestra.py]: What does `zone_confirmation` tell the user? If you really wanted plan_name_message to have two different values, might be better to have the message sent as `"plan_name=plan_name&zone_confirmation=value`

**Comment by khanguslee** [comment]
Please make codacy happy. Looks like line 117 has a trailing whitespace

File: orchestra.py (hunk 1 above)
Comment (chamaka1) [reply orchestra.py]: plan_generated is to say a plan was generated so it is displayed on the website and plan_name is for the overlay

File: orchestra.py (hunk 2 above)
Comment (chamaka1) [reply orchestra.py]: zone confirmation is a list of data specifying the distance where there is a change in power and its essential information cause you wont be able to tell from the plan name if we have many plans

Comment (khanguslee) [reply orchestra.py]: Hmm okay. Well can you please change the format of the message to above? We would like to keep stuff like `&key=value`

Comment (khanguslee) [inline orchestra.py]: Change this to:
```
plan_name_message = "plan_name= " + str(plan_name) + "&zone_confirmation=" + zone_confirmation
```

File: orchestra.py [hunk 3]
```diff
@@ -109,6 +108,16 @@ def on_message(client, userdata, msg):
...
         else:
             print("process mqtt_max_speed running")
+            
+        zone_interpolation_file_path = os.path.join(os.path.dirname(os.path.realpath(sys.argv[0])), "run_plans", plan_name)
+        
+        with open(zone_interpolation_file_path, 'rb') as y:
+            interps_loaded = pickle.load(y)
+            
+        zone_confirmation = str(interps_loaded[2])
+        plan_name_message = "plan_name= " + str(plan_name) + "-" + zone_confirmation
```
Comment (khanguslee) [inline orchestra.py]: As above

**Comment by khanguslee** [comment]
Please have a check of https://app.codacy.com/app/mhp-admin/MHP_Modelling/pullRequest?prid=3982462

You've added more warnings/errors.

File: orchestra.py [hunk 4]
```diff
@@ -109,6 +108,16 @@ def on_message(client, userdata, msg):
             
         else:
             print("process mqtt_max_speed running")
+            
+        zone_interpolation_file_path = os.path.join(os.path.dirname(os.path.realpath(sys.argv[0])), "run_plans", plan_name)
```
Comment (harsilspatel) [inline orchestra.py]: Yes! Good work on getting realpath of the file

**Comment by harsilspatel** [comment]
Also, I think we should merge this script with the orchestrator, the one that's getting and setting the overlays? 
What do we think @khanguslee @hallgchris?

**Comment by khanguslee** [comment]
> Also, I think we should merge this script with the orchestrator, the one that's getting and setting the overlays?
> What do we think @khanguslee @hallgchris?

Perhaps a much cleaner method of handling the whole bike ecosystem is to create V1, V2, V3 etc. repositories that then take MHP_Modelling, MHP_Raspicam, MHP_DAS, MHP-DAS-Web-Server as sub modules? We can still have the individual orchestrator scripts in each repo, but to run it on our bike we can create a single orchestrator that then listens to everything.

File: rider_plan_finder.py [hunk 5]
```diff
@@ -32,38 +32,28 @@
     data_store = json.load(f)
     
 # name of file to be saved
-run_plan_name = "default.pkl"
```
Comment (khanguslee) [inline rider_plan_finder.py]: Thanks for cleaning everything up :D

**Pull request merged**